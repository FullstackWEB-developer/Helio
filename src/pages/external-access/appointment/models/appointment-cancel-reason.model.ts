export interface AppointmentCancelReason {
    appointmentCancelReasonId: number;
    name: string;
    noShow: boolean;
    patientRescheduled: boolean;
    providerUnavailable: boolean;
    slotAvailable: boolean;
}
