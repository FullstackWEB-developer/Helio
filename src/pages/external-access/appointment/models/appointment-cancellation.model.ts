export interface AppointmentCancellationModel {
    patientId: number;
    appointmentCancelReasonId: number;
    cancellationReason: string;
}
