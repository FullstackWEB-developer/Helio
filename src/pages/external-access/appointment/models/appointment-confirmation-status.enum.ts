export enum ConfirmationStatus {
    ConfirmedViaSMS = 161,
    ConfirmedViaEmail = 121
}