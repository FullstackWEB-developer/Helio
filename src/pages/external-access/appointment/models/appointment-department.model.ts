export interface AppointmentDepartmentModel {
    latLong: number[];
    name: string;
    address: string;
    address2?: string;
    city: string;
    state: string;
    zip: string;
}
