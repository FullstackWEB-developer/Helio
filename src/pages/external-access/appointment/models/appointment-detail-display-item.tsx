export interface AppointmentDetailDisplayItem {
    title: string,
    description: string
}
