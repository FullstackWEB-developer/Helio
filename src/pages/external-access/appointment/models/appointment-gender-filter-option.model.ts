
export enum GenderFilterOption {
    Female = 'GENDER_FEMALE',
    Male = 'GENDER_MALE'
}
