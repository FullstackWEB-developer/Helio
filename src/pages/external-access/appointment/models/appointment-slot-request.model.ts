export enum AppointmentSlotTimeOfDay {
    EarlyMorning = 1,
    Morning = 2,
    Afternoon = 3
}
export interface AppointmentSlotRequest {
    departmentId?: number;
    startDate: Date;
    endDate: Date;
    providerId?: number[];
    ignoreSchedulablePermission?: boolean;
    appointmentTypeId: number;
    itemCount?: number;
    patientId?: number;
    allowMultipleDepartment?: boolean;
    patientDefaultDepartmentId?: number;
    patientDefaultProviderId?: number
    timeOfDays?: AppointmentSlotTimeOfDay[];
    firstAvailable: boolean;
}

