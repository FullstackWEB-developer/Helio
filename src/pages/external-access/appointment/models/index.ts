export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
