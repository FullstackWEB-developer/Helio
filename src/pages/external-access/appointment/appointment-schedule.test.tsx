import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {act, render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import AppointmentSchedule from './appointment-schedule';
import Router from "react-router";
import Api from '@shared/services/api';
import { VerifiedPatient } from '@pages/patients/models/verified-patient';
import { Location, Provider } from '@shared/models';
import { Appointment, AppointmentType } from './models';
import { PatientAppointmentType } from '@shared/models/patient-appointment-type.enum';
jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useParams: jest.fn().mockReturnValue({ type: "ticket-department" }),
  }))
  jest.setTimeout(70000);
describe("AppointmentSchedule tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {
            providerList: [
                {
                    ansiNameCode: '',
                    ansiSpecialtyCode: '',
                    billable: false,
                    createEncounterOnCheckin: false,
                    firstName: 'provider',
                    lastName: 'lastname',
                    id: 3
                }
            ] as Provider[],
            locationList: [{
                medicationHistoryConsent: true,
                timeZoneOffset: 1,
                isHospitalDepartment: true,
                state: "string",
                portalUrl: "string",
                city: "string",
                placeOfServiceFacility: true,
                serviceDepartment: true,
                latitude: 1,
                fax: "string",
                doesNotObservedSt: true,
                parkingInformation: "string",
                id: 1,
                address: "string",
                placeOfServiceTypeId: 1,
                longitude: 1,
                clinicals: "string",
                timeZone: 1,
                patientDepartmentName: "string",
                chartSharingGroupId: "string",
                name: "string",
                placeOfServiceTypeName: "string",
                phone: "string",
                address2: "string",
                clinicalProviderFax: "string",
                ecommerceCreditCardTypes: ["1", "2"],
                zip: "string",
                timeZoneName: "string"
            }
            ] as Location[]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        callsLogState: {
            isFiltered: false
        },
        ticketState: {
            lookupValues: []
        },
        patientsState: {
            verifiedPatient: {
                firstName: "string",
                lastName: "string",
                patientId: 123,
                primaryProviderId: 12,
                departmentId: 1,
                defaultProviderId: 12,
                defaultDepartmentId: 1,
                dateOfBirth: new Date("1995-11-15"),
                token: "string",
                tokenExpiration: new Date("2023-11-15")
            } as VerifiedPatient
        },
        externalAccessState: {
            appointmentsState: {
                patientUpcomingAppointment: {
                    appointmentId: "string",
                    appointmentStatus: "Future",
                    appointmentType: "string",
                    appointmentTypeId:1,
                    copay:1,
                    date: new Date("2022-11-15"),
                    departmentId: 1,
                    duration:1,
                    patientAppointmentTypeName: "string",
                    providerId:1,
                    startTime: "string",
                    startDateTime: new Date("2022-11-15"),
                } as Appointment
            }
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders AppointmentSchedule correctly", async () => {
        let fragment: any;
        jest.spyOn(Api, 'get').mockImplementation((url) => {
            if (url.includes('appointmenttypes/patient-appointment-types')) {
                return new Promise((resolve) => {
                    resolve({
                        data: [{
                            id: 10,
                            instructions: "string",
                            name: "string",
                            cancelable: true,
                            cancelationTimeFrame: 10,
                            cancelationFee: 10,
                            reschedulable: true,
                            rescheduleTimeFrame: 10,
                            description: "string",
                            createdByName: "string",
                            modifiedByName: "string",
                            createdOn: new Date("2022-11-15"),
                            modifiedOn: new Date("2022-11-15"),
                            selectableByPatient: true,
                            selectedProviders: [3],
                            patientType: PatientAppointmentType.All
                        }] as AppointmentType[]
                    })
                })
            } else if(url.includes('/appointments')){
                return new Promise((resolve) => {
                    resolve({
                        data: [{
                            appointmentId: "string",
                            appointmentStatus: "Future",
                            appointmentType: "string",
                            appointmentTypeId:1,
                            copay:1,
                            date: new Date("2022-11-15"),
                            departmentId: 1,
                            duration:1,
                            patientAppointmentTypeName: "string",
                            providerId:1,
                            startTime: "string",
                            startDateTime: new Date("2022-11-15"),
                        }] as Appointment[]
                    })
                })
            } else {
                return new Promise((resolve) => {
                    resolve({
                        data: [{
                            stateCode: 'AL',
                            name: 'Alabama'
                        }]
                    })
                })
            }
        });
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <AppointmentSchedule/>          
            </TestWrapper>);

            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders AppointmentSchedule correctly api return empty array", async () => {
        let fragment: any;
        jest.spyOn(Api, 'get').mockImplementation((url) => {
            if (url.includes('appointmenttypes/patient-appointment-types')) {
                return new Promise((resolve) => {
                    resolve({
                        data: [] as AppointmentType[]
                    })
                })
            } else if(url.includes('/appointments')){
                return new Promise((resolve) => {
                    resolve({
                        data: [] as Appointment[]
                    })
                })
            } else {
                return new Promise((resolve) => {
                    resolve({
                        data: [{
                            stateCode: 'AL',
                            name: 'Alabama'
                        }]
                    })
                })
            }
        });
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <AppointmentSchedule/>          
            </TestWrapper>);

            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });
})
