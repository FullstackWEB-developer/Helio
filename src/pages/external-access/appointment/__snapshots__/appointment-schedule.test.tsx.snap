// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentSchedule tests renders AppointmentSchedule correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <h4
      class="pb-9"
    >
      external_access.schedule_appointment.what_you_are_coming_for
    </h4>
    <div
      class="pb-10"
    >
      external_access.schedule_appointment.what_you_are_coming_for_desc
    </div>
    <div
      class="w-88"
    >
      <form
        novalidate=""
      >
        <div
          class="select-wrapper relative w-full flex flex-col h-20 "
        >
          <div
            class="select relative flex flex-col "
          >
            <input
              autocomplete="on"
              autocorrect="on"
              class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
              data-testid="appointment_type"
              name="appointment_type"
              spellcheck="true"
              type="text"
              value=""
            />
            <label
              class="select-label absolute truncate"
            >
              <span
                class="text-danger"
              >
                *
              </span>
              <span
                class="select-label-span body2"
              >
                external_access.schedule_appointment.appointment_type
              </span>
            </label>
            <div
              class="absolute pt-4 right-4 "
            >
              <div
                class="flex flex-row items-start"
              >
                <div
                  class="cursor-pointer"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="select-arrow-fill"
                        d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                        transform="translate(5 16.273) rotate(-90)"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="absolute py-2 options"
            >
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full py-2"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-string"
                    data-testid="select-cell-text-string"
                  >
                    string
                  </div>
                  <span
                    class="pl-4 body3-small assistive-text select-cell-assistive-text-line-clamped"
                  >
                    string
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="select-wrapper relative w-full flex flex-col h-20 "
        >
          <div
            class="select relative flex flex-col "
          >
            <input
              autocomplete="on"
              autocorrect="on"
              class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
              data-testid="provider"
              name="provider"
              spellcheck="true"
              type="text"
              value=""
            />
            <label
              class="select-label absolute truncate"
            >
              <span
                class="select-label-span body2"
              >
                external_access.schedule_appointment.provider
              </span>
            </label>
            <div
              class="absolute pt-4 right-4 "
            >
              <div
                class="flex flex-row items-start"
              >
                <div
                  class="cursor-pointer"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="select-arrow-fill"
                        d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                        transform="translate(5 16.273) rotate(-90)"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="absolute py-2 options"
            >
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-common.any"
                    data-testid="select-cell-text-common.any"
                  >
                    common.any
                  </div>
                </div>
              </div>
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-provider lastname"
                    data-testid="select-cell-text-provider lastname"
                  >
                    provider lastname
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="select-wrapper relative w-full flex flex-col h-20 "
        >
          <div
            class="select relative flex flex-col "
          >
            <input
              autocomplete="on"
              autocorrect="on"
              class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
              data-testid="location"
              name="location"
              spellcheck="true"
              type="text"
              value=""
            />
            <label
              class="select-label absolute truncate"
            >
              <span
                class="select-label-span body2"
              >
                external_access.schedule_appointment.location
              </span>
            </label>
            <div
              class="absolute pt-4 right-4 "
            >
              <div
                class="flex flex-row items-start"
              >
                <div
                  class="cursor-pointer"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="select-arrow-fill"
                        d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                        transform="translate(5 16.273) rotate(-90)"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="absolute py-2 options"
            >
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-common.any"
                    data-testid="select-cell-text-common.any"
                  >
                    common.any
                  </div>
                </div>
              </div>
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-String"
                    data-testid="select-cell-text-String"
                  >
                    String
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pt-6"
        >
          <div
            class="pb-5 body2"
          >
            external_access.schedule_appointment.select_preference
          </div>
          <div
            class=""
          >
            <div
              class="radio h-9 "
            >
              <input
                checked=""
                data-testid="time_preference_1"
                id="time_preference_1"
                name="time_preference"
                type="radio"
                value="1"
              />
              <label
                class="body2  "
                for="time_preference_1"
              >
                <div
                  class="truncate relative w-full h-full inline-block"
                >
                  <span
                    class="body2"
                  >
                    external_access.schedule_appointment.first_available
                  </span>
                </div>
              </label>
            </div>
            <div
              class="radio h-9 "
            >
              <input
                data-testid="time_preference_2"
                id="time_preference_2"
                name="time_preference"
                type="radio"
                value="2"
              />
              <label
                class="body2  "
                for="time_preference_2"
              >
                <div
                  class="truncate relative w-full h-full inline-block"
                >
                  <span
                    class="body2"
                  >
                    external_access.schedule_appointment.select_preferred
                  </span>
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          class="flex justify-start pt-8"
        >
          <button
            class="big-button"
            disabled=""
            type="submit"
          >
            <div
              class="flex flex-row justify-center items-center space-x-2"
            >
              <div>
                external_access.schedule_appointment.find_appointment
              </div>
            </div>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppointmentSchedule tests renders AppointmentSchedule correctly api return empty array 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <h4
      class="pb-9"
    >
      external_access.schedule_appointment.what_you_are_coming_for
    </h4>
    <div
      class="pb-10"
    >
      external_access.schedule_appointment.what_you_are_coming_for_desc
    </div>
    <div
      class="w-88"
    >
      <form
        novalidate=""
      >
        <div
          class="select-wrapper relative w-full flex flex-col h-20 "
        >
          <div
            class="select relative flex flex-col "
          >
            <input
              autocomplete="on"
              autocorrect="on"
              class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
              data-testid="appointment_type"
              name="appointment_type"
              spellcheck="true"
              type="text"
              value=""
            />
            <label
              class="select-label absolute truncate"
            >
              <span
                class="text-danger"
              >
                *
              </span>
              <span
                class="select-label-span body2"
              >
                external_access.schedule_appointment.appointment_type
              </span>
            </label>
            <div
              class="absolute pt-4 right-4 "
            >
              <div
                class="flex flex-row items-start"
              >
                <div
                  class="cursor-pointer"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="select-arrow-fill"
                        d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                        transform="translate(5 16.273) rotate(-90)"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="absolute py-2 options"
            >
              <div
                class="w-full pt-2 text-center subtitle3-small"
              >
                common.autocomplete_search
              </div>
            </div>
          </div>
        </div>
        <div
          class="select-wrapper relative w-full flex flex-col h-20 "
        >
          <div
            class="select relative flex flex-col "
          >
            <input
              autocomplete="on"
              autocorrect="on"
              class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
              data-testid="provider"
              name="provider"
              spellcheck="true"
              type="text"
              value=""
            />
            <label
              class="select-label absolute truncate"
            >
              <span
                class="select-label-span body2"
              >
                external_access.schedule_appointment.provider
              </span>
            </label>
            <div
              class="absolute pt-4 right-4 "
            >
              <div
                class="flex flex-row items-start"
              >
                <div
                  class="cursor-pointer"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="select-arrow-fill"
                        d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                        transform="translate(5 16.273) rotate(-90)"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="absolute py-2 options"
            >
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-common.any"
                    data-testid="select-cell-text-common.any"
                  >
                    common.any
                  </div>
                </div>
              </div>
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-provider lastname"
                    data-testid="select-cell-text-provider lastname"
                  >
                    provider lastname
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="select-wrapper relative w-full flex flex-col h-20 "
        >
          <div
            class="select relative flex flex-col "
          >
            <input
              autocomplete="on"
              autocorrect="on"
              class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
              data-testid="location"
              name="location"
              spellcheck="true"
              type="text"
              value=""
            />
            <label
              class="select-label absolute truncate"
            >
              <span
                class="select-label-span body2"
              >
                external_access.schedule_appointment.location
              </span>
            </label>
            <div
              class="absolute pt-4 right-4 "
            >
              <div
                class="flex flex-row items-start"
              >
                <div
                  class="cursor-pointer"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="select-arrow-fill"
                        d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                        transform="translate(5 16.273) rotate(-90)"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="absolute py-2 options"
            >
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-common.any"
                    data-testid="select-cell-text-common.any"
                  >
                    common.any
                  </div>
                </div>
              </div>
              <div
                class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
              >
                <div
                  class="flex flex-col justify-center w-full"
                  data-test-id="select-cell-icon-content"
                >
                  <div
                    class="flex items-center pl-4"
                    data-test-id="select-cell-text-String"
                    data-testid="select-cell-text-String"
                  >
                    String
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pt-6"
        >
          <div
            class="pb-5 body2"
          >
            external_access.schedule_appointment.select_preference
          </div>
          <div
            class=""
          >
            <div
              class="radio h-9 "
            >
              <input
                checked=""
                data-testid="time_preference_1"
                id="time_preference_1"
                name="time_preference"
                type="radio"
                value="1"
              />
              <label
                class="body2  "
                for="time_preference_1"
              >
                <div
                  class="truncate relative w-full h-full inline-block"
                >
                  <span
                    class="body2"
                  >
                    external_access.schedule_appointment.first_available
                  </span>
                </div>
              </label>
            </div>
            <div
              class="radio h-9 "
            >
              <input
                data-testid="time_preference_2"
                id="time_preference_2"
                name="time_preference"
                type="radio"
                value="2"
              />
              <label
                class="body2  "
                for="time_preference_2"
              >
                <div
                  class="truncate relative w-full h-full inline-block"
                >
                  <span
                    class="body2"
                  >
                    external_access.schedule_appointment.select_preferred
                  </span>
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          class="flex justify-start pt-8"
        >
          <button
            class="big-button"
            disabled=""
            type="submit"
          >
            <div
              class="flex flex-row justify-center items-center space-x-2"
            >
              <div>
                external_access.schedule_appointment.find_appointment
              </div>
            </div>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
